src/main/java/io/vertx/ext/couchdb/impl/CouchdbClientImpl.java

```diff
@@ -81,26 +74,57 @@ public Future<Buffer> rawCall(JsonObject params) {
     initClient();
     HttpRequest<Buffer> request = client.request(method, path);
 
-    if (this.credentials != null) {
-      request.authentication(this.credentials);
+    if (clientOptions.getCredentials() != null) {
+      request.authentication(clientOptions.getCredentials().toAuthCredentials());
     }
     request.send()
-        .onFailure(promise::fail)
-        .onSuccess(response -> promise.complete(response.body()));
+      .onFailure(promise::fail)
+      .onSuccess(response -> promise.complete(response.body()));
 
     return promise.future();
   }
 
   private void initClient() {
     if (client == null) {
-      client = WebClient.create(vertx, options);
+      client = WebClient.create(vertx, clientOptions.getWebClientOptions());
     }
   }
 
   @Override
-  public CouchdbClient authetication(Credentials credentials) {
-    this.credentials = credentials;
-    return this;
+  public Future<JsonObject> createDb(JsonObject options) {
+    Promise<JsonObject> promise = Promise.promise();
+    var databaseName = options.getString("db_name");
+    var shards = options.getInteger("q", 8);
+    var replicas = options.getInteger("n", 3);
+    var partitioned = options.getBoolean("partitioned", false);
+
+    initClient();
+    HttpRequest<Buffer> request = client.put("/" + databaseName)
+      .addQueryParam("q", shards.toString())
+      .addQueryParam("n", replicas.toString())
+      .addQueryParam("partitioned", partitioned.toString());
+
+    if (clientOptions.getCredentials() != null) {
+      request.authentication(clientOptions.getCredentials().toAuthCredentials());
+    }
+
+    request.send()
+      .onSuccess(response -> {
+        int statusCode = response.statusCode();
+        JsonObject result = response.bodyAsJsonObject();
+        if (statusCode == 201 || statusCode == 202) {
+          promise.complete(result);
+        } else {
+          promise.fail(new RuntimeException(
```

Comment:
We might want our own exception that can carry the returned body

Reply:
Noted